monai/transforms/intensity/array.py

```diff
@@ -297,6 +298,91 @@ def __call__(self, img: np.ndarray) -> np.ndarray:
         return scaler(img)
 
 
+class RandBiasField(RandomizableTransform):
+    """
+    Random bias field augmentation for MR images.
+    The bias field is considered as a linear combination of smoothly varying basis (polynomial)
+    functions, as described in `Automated Model-Based Tissue Classification of MR Images of the Brain
+    <https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=811270>`_.
+    This implementation adapted from `NiftyNet
+    <https://github.com/NifTK/NiftyNet>`_.
+    Referred to `Longitudinal segmentation of age-related white matter hyperintensities
+    <https://www.sciencedirect.com/science/article/pii/S1361841517300257?via%3Dihub>`_.
+
+    Args:
+        degree: degree of freedom of the polynomials. The value should be no less than 1.
+            Defaults to 3.
+        coeff_range: range of the random coefficients. Defaults to (0.0, 0.1).
+        dtype: output data type, defaut to float32.
+        prob: probability to do random bias field.
+
+    """
+
+    def __init__(
+        self,
+        degree: int = 3,
+        coeff_range: Tuple[float, float] = (0.0, 0.1),
+        dtype: DtypeLike = np.float32,
+        prob: float = 1.0,
+    ) -> None:
+        RandomizableTransform.__init__(self, prob)
+        self.degree = degree
+        self.coeff_range = coeff_range
+        self.dtype = dtype
+
+    def _generate_random_field(
+        self,
+        spatial_dims: Tuple[int, ...],
+        degree: int,
+        coeff_range: Tuple[float, float] = (0.0, 0.1),
+    ):
+        """
+        products of polynomials as bias field estimations
+        """
+        if degree < 1:
+            raise ValueError("degree should be no less than 1.")
+        rank = len(spatial_dims)
+        coeff_mat = np.zeros((degree + 1,) * rank)
+        n_coeff = int(np.prod([(degree + k) / k for k in range(1, rank + 1)]))
+        coeff = self.R.uniform(*coeff_range, n_coeff)
+        coords = [np.linspace(-1.0, 1.0, dim, dtype=np.float32) for dim in spatial_dims]
+        if rank == 2:
+            coeff_mat[np.tril_indices(degree + 1)] = coeff
+            field = np.polynomial.legendre.leggrid2d(coords[0], coords[1], coeff_mat)
+        elif rank == 3:
+            pts: List[List[int]] = [[0, 0, 0]]
+            for i in range(degree + 1):
+                for j in range(degree + 1 - i):
+                    for k in range(degree + 1 - i - j):
+                        pts.append([i, j, k])
+            if len(pts) > 1:
+                pts = pts[1:]
+            np_pts = np.stack(pts)
+            coeff_mat[np_pts[:, 0], np_pts[:, 1], np_pts[:, 2]] = coeff
+            field = np.polynomial.legendre.leggrid3d(coords[0], coords[1], coords[2], coeff_mat)
+        else:
+            raise NotImplementedError("only supoprts 2D or 3D fields")
+        return field
+
+    def randomize(self, data: Optional[Any] = None) -> None:
+        super().randomize(None)
```

Comment:
```suggestion
    def randomize(self, spatial_dims: Optional[Any] = None) -> None:
        super().randomize(None)
        self._bias_fields = np.stack(
            [self._generate_random_field(spatial_dims, self.degree, self.coeff_range) for _ in range(num_channels)],
            axis=0,
        )
```

could follow `RandAxisFlip` as an example for the randomize() method: https://github.com/Project-MONAI/MONAI/blob/master/monai/transforms/spatial/array.py#L781-L809